### PR TITLE
Create components/metrics subtree.

### PR DESCRIPTION
**Pull request by briantting**
Create components/metrics subtree from https://chromium.googlesource.com/chromium/src/components/metrics at commit hash 734293972a4b3f6d949e489ac02145538fb59c70.

b/279965125

### CODE REVIEW

File: components/metrics/METADATA [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+description:
+  "Subtree at components/metrics."
+
+third_party {
+  url {
+    type: GIT
+    value: "https://chromium.googlesource.com/chromium/src/components/metrics"
+  }
+  version: "734293972a4b3f6d949e489ac02145538fb59c70"
+  last_upgrade_date {
```
Comment (andrewsavage1) [inline components/metrics/METADATA:11]: We want to import it from 2018 instead of a more recent version?

**Review by joeltine — CHANGES_REQUESTED** [review]
These three files fail the "end of file" lint check:

-  components/metrics/expired_histograms_checker_unittest.cc
-  components/metrics/expired_histogram_util.h
-  components/metrics/expired_histogram_util.cc

If you check these in, every engineer will get warnings about these when doing `git push` (and moreover, the files will get auto-fixed corrupting the push). Can we add these to an ignore list somewhere?

File: components/metrics/METADATA (hunk 1 above)
Comment (joeltine) [reply components/metrics/METADATA:11]: Yes as it has to match our base/metrics package, which itself is from 2018.

Comment (joeltine) [reply components/metrics/METADATA:11]: and third_party/metric_proto

**Comment by briantting** [comment]
> These three files fail the "end of file" lint check:
> 
> * components/metrics/expired_histograms_checker_unittest.cc
> * components/metrics/expired_histogram_util.h
> * components/metrics/expired_histogram_util.cc
> 
> If you check these in, every engineer will get warnings about these when doing `git push` (and moreover, the files will get auto-fixed corrupting the push). Can we add these to an ignore list somewhere?

We can just fix these in a PR after the merge.

**Comment by briantting** [comment]
```
components/metrics/expired_histogram_util.cc:18:  Add #include <string> for string  [build/include_what_you_use] [4]
components/metrics/expired_histogram_util.cc:28:  Add #include <memory> for make_unique<>  [build/include_what_you_use] [4]
Done processing components/metrics/expired_histogram_util.cc
components/metrics/expired_histograms_checker_unittest.cc:31:  Add #include <string> for string  [build/include_what_you_use] [4]
Done processing components/metrics/expired_histograms_checker_unittest.cc

```

the 3 files have more than just end of file lint errors annoyingly

**Pull request merged**